src/connector/src/parser/debezium/avro_parser.rs

```diff
@@ -35,34 +35,27 @@ use crate::schema::schema_registry::{
 
 #[derive(Debug)]
 pub struct DebeziumAvroAccessBuilder {
-    schema: ResolvedAvroSchema,
+    reader_schema: ResolvedAvroSchema,
     schema_resolver: Arc<ConfluentSchemaCache>,
-    key_schema: Option<Arc<Schema>>,
     value: Option<Value>,
-    encoding_type: EncodingType,
 }
 
-// TODO: reduce encodingtype match
 impl AccessBuilder for DebeziumAvroAccessBuilder {
     async fn generate_accessor(
         &mut self,
         payload: Vec<u8>,
         _: &crate::source::SourceMeta,
     ) -> ConnectorResult<AccessImpl<'_>> {
         let (schema_id, mut raw_payload) = extract_schema_id(&payload)?;
-        let schema = self.schema_resolver.get_by_id(schema_id).await?;
-        self.value = Some(from_avro_datum(schema.as_ref(), &mut raw_payload, None)?);
-        self.key_schema = match self.encoding_type {
-            EncodingType::Key => Some(schema),
-            EncodingType::Value => None,
-        };
+        let writer_schema = self.schema_resolver.get_by_id(schema_id).await?;
+        self.value = Some(from_avro_datum(
+            writer_schema.as_ref(),
+            &mut raw_payload,
+            Some(&self.reader_schema.original_schema),
+        )?);
         Ok(AccessImpl::Avro(AvroAccess::new(
-            self.value.as_mut().unwrap(),
-            // Assumption: Key will not contain reference, so unresolved schema can work here.
-            AvroParseOptions::create(match self.encoding_type {
-                EncodingType::Key => self.key_schema.as_mut().unwrap(),
-                EncodingType::Value => &self.schema.original_schema,
```

Comment:
The key uses writer schema, with reader schema dropped by `..` below; while the value uses reader schema.